data/keys

```diff
@@ -259,35 +259,106 @@ Global {
 	KeyPress = "Up" { Actions = "MoveVertical -10" }
 	KeyPress = "Down" { Actions = "MoveVertical 10" }
+
 	Keypress = "Shift Left" { Actions = "MoveHorizontal -1" }
 	Keypress = "Shift Right" { Actions = "MoveHorizontal 1" }
 	Keypress = "Shift Up" { Actions = "MoveVertical -1" }
 	Keypress = "Shift Down" { Actions = "MoveVertical 1" }
-	Keypress = "Mod4 Left" { Actions = "ResizeHorizontal -10" }
-	Keypress = "Mod4 Right" { Actions = "ResizeHorizontal 10" }
-	Keypress = "Mod4 Up" { Actions = "ResizeVertical -10" }
-	Keypress = "Mod4 Down" { Actions = "ResizeVertical 10" }
+
+	KeyPress = "Ctrl Left" { Actions = "MoveHorizontal -10%" }
+	KeyPress = "Ctrl Right" { Actions = "MoveHorizontal 10%" }
+	KeyPress = "Ctrl Up" { Actions = "MoveVertical -10%" }
+	KeyPress = "Ctrl Down" { Actions = "MoveVertical 10%" }
+
+	Keypress = "Ctrl Shift Left" { Actions = "MoveHorizontal -1%" }
+	Keypress = "Ctrl Shift Right" { Actions = "MoveHorizontal 1%" }
+	Keypress = "Ctrl Shift Up" { Actions = "MoveVertical -1%" }
+	Keypress = "Ctrl Shift Down" { Actions = "MoveVertical 1%" }
+
 	Keypress = "Mod1 Left" { Actions = "ResizeHorizontal -10" }
 	Keypress = "Mod1 Right" { Actions = "ResizeHorizontal 10" }
 	Keypress = "Mod1 Up" { Actions = "ResizeVertical -10" }
 	Keypress = "Mod1 Down" { Actions = "ResizeVertical 10" }
-	Keypress = "Shift Mod4 Left" { Actions = "ResizeHorizontal -1" }
-	Keypress = "Shift Mod4 Right" { Actions = "ResizeHorizontal 1" }
-	Keypress = "Shift Mod4 Up" { Actions = "ResizeVertical -1" }
-	Keypress = "Shift Mod4 Down" { Actions = "ResizeVertical 1" }
+
 	Keypress = "Shift Mod1 Left" { Actions = "ResizeHorizontal -1" }
 	Keypress = "Shift Mod1 Right" { Actions = "ResizeHorizontal 1" }
 	Keypress = "Shift Mod1 Up" { Actions = "ResizeVertical -1" }
 	Keypress = "Shift Mod1 Down" { Actions = "ResizeVertical 1" }
+
+	Keypress = "Ctrl Mod1 Left" { Actions = "Ctrl ResizeHorizontal -10%" }
+	Keypress = "Ctrl Mod1 Right" { Actions = "Ctrl ResizeHorizontal 10%" }
+	Keypress = "Ctrl Mod1 Up" { Actions = "Ctrl ResizeVertical -10%" }
+	Keypress = "Ctrl Mod1 Down" { Actions = "Ctrl ResizeVertical 10%" }
+
+	Keypress = "Ctrl Shift Mod1 Left" { Actions = "Ctrl ResizeHorizontal -1%" }
+	Keypress = "Ctrl Shift Mod1 Right" { Actions = "Ctrl ResizeHorizontal 1%" }
+	Keypress = "Ctrl Shift Mod1 Up" { Actions = "Ctrl ResizeVertical -1%" }
+	Keypress = "Ctrl Shift Mod1 Down" { Actions = "Ctrl ResizeVertical 1%" }
+
+	Keypress = "Mod4 Left" { Actions = "ResizeHorizontal -10" }
+	Keypress = "Mod4 Right" { Actions = "ResizeHorizontal 10" }
+	Keypress = "Mod4 Up" { Actions = "ResizeVertical -10" }
+	Keypress = "Mod4 Down" { Actions = "ResizeVertical 10" }
+
+	Keypress = "Shift Mod4 Left" { Actions = "ResizeHorizontal -1" }
+	Keypress = "Shift Mod4 Right" { Actions = "ResizeHorizontal 1" }
+	Keypress = "Shift Mod4 Up" { Actions = "ResizeVertical -1" }
+	Keypress = "Shift Mod4 Down" { Actions = "ResizeVertical 1" }
+
+	KeyPress = "Q" { Actions = "MoveToEdge TopLeft" }
```

Comment:
I just realized that moveresize commands are special and ones like MoveToEdge or SetGeometry won't work in MoveResize block (or ResizeHorizontal won't work outside MoveResize block).

Would be be acceptable to unify the code so that any move/resize command can work inside or outside of MoveResize block the same way? From the look of it, I think this is possible, but I wanted to check first before doing anything.

Reply:
I've started a batch of refactoring enabling pekwm to handle other X events while performing a move/resize, keyboard move resize is done.

My plan was to update the EventHandler so that it can return an action back to the caller, and then add to the code to sync-up the geometry.

Not sure on how it will fit with the current design or if it is better to do the actual processing inside of the EventHandler.

Anyway, adding support for more actions inside of keyboard move resize seems like a good idea at least for the SetGeometry.

Reply:
Thanks. I'll do nothing then. Just let me know if you need a hand :)

Reply:
@pclouds if you want to have a go at it feel very free! There's enough things to get done! That was my plan on how to achieve what you were thinking of.